### ISSUE TITLE
vso cli does not allow sign in with personal account

### ISSUE THREAD

**Issue by JtMotoX** [MicrosoftDocs/vscodespaces#394]
OS: **macOS 10.15.3**
VSO Cli Version: **1.0.1954**

I am unable to link vso cli to my personal microsoft account.

```
$ vso start -v
[2020-04-09 21:23:52.841 CLI I] vso v1.0.1954.49374 (pid: 77574)
[2020-04-09 21:23:53.275 CLI I] Authenticating...
To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code *** to authenticate.
```

I navigate to the URL mentioned above and enter the code.  I then attempt to log in with my personal account and I am stopped with the following error:

> You can't sign in here with a personal account. Use your work or school account instead.

**Comment by jmbockhorst** [comment]
Running into this issue as well on linux and version **1.0.1961**.

**Comment by definitelycarter** [comment]
I'm hitting this issue as well on linux and **1.0.1961**.

**Comment by taiyosogawa** [comment]
@JtMotoX, good to hear from you again. Thank you all for reporting this.

We're taking a look into this issue. Could you confirm whether or not you are using an account linked to an Azure account? Even though they aren't billed, self-hosted environments require a [VS Online Plan](https://docs.microsoft.com/en-us/visualstudio/online/how-to/self-hosting-cli#register-your-machine), so the account you log in would have to match the one that has access to that plan.

**Comment by JtMotoX** [comment]
@taiyosogawa, Thanks for looking into this.  I wasn't expecting a response this fast.

Not sure how to tell if it's linked to an Azure account.  When I sign into my personal account and navigate to [online.visualstudio.com](https://online.visualstudio.com/) I am able to see my plan (vso-plan-westus2).  This is the same account and plan that I used for linking my local VSCode environment.

**Comment by taiyosogawa** [comment]
@JtMotoX, Thanks for confirming. If you're able to see the plans on online.visualstudio.com, it sounds like you're doing everything right. We'll work on resolving the issue. 

In the meantime, if you're looking to use the new self-hosted service/daemon feature and have access to VS Code on the host machine, that should work: https://docs.microsoft.com/en-us/visualstudio/online/how-to/self-hosting-vscode

I'll update when we determine why the CLI authentication is not working.

**Comment by definitelycarter** [comment]
> @JtMotoX, Thanks for confirming. If you're able to see the plans on online.visualstudio.com, it sounds like you're doing everything right. We'll work on resolving the issue.
> 
> In the meantime, if you're looking to use the new self-hosted service/daemon feature and have access to VS Code on the host machine, that should work: https://docs.microsoft.com/en-us/visualstudio/online/how-to/self-hosting-vscode
> 
> I'll update when we determine why the CLI authentication is not working.

Steps to reproduce:

1. When you run `vso start` it asks you to go to https://microsoft.com/devicelogin

![image](https://user-images.githubusercontent.com/6287039/78948325-e0e45e80-7a95-11ea-9034-e5d10b2139be.png)

2. Enter the code and click next.

3. Enter your email and click next.  If it is a personal account you will receive the error below

![image](https://user-images.githubusercontent.com/6287039/78948408-33257f80-7a96-11ea-84b7-9ba8c67c188f.png)

**Note** http://login.live.com does let me use my personal email to login; However, the URL in the screenshot above does not.

**Comment by taiyosogawa** [comment]
@definitelycarter, thanks for putting together your repro steps. Screenshots are always super helpful :)

We've determined the problem and are working on the fix. Until it is published, login will fail with a personal account. We'll update as soon as the patched version is available.

**Comment by Niek** [comment]
The same happens when it's not a personal account, but an account with a custom email address (domain) it seems. I get this error:

<img width="414" alt="Screen Shot 2020-04-14 at 10 45 46" src="https://user-images.githubusercontent.com/213140/79204667-34abca80-7e3d-11ea-94bd-50c2f5b30469.png">

**Comment by yume-chan** [comment]
Seemingly the issue has been fixed, but the homebrew formula for macOS has not been updated.

I managed to get it working by

1. Download a newer version from https://vsoagentdownloads.blob.core.windows.net/vsoagent/VSOAgent_osx_3633968.zip (version number got from the official Windows installation script https://aka.ms/install-vso-windows, it will redirect to a url with a version number)
2. In Finder, double click to extract the zip file
3. In Terminal, cd to the extracted folder, execute the following (to make them executable)
    ```sh
    chmod +x vso
    chmod +x vsls-agent
    ```
4. In Finder, right click "vso", click "Open", in the popup, click "Open" (to unlock the file)
5. Repeat 4 for "vsls-agent"
6. In Terminal, run `./vso"

**Comment by davide-rota** [comment]
I can confirm that on Ubuntu 18.04 `vso cli` linked successfully the environment with my MSA account (@live.com) 👍

**Comment by gusbicalho** [comment]
I'm trying to set up vso on a remote machine using Ubuntu.

```
$ vso --version
[2020-04-14 12:51:14.475 CLI I] vso v1.0.1998.63218 (pid: 29338)
Command-line interface for Visual Studio Online
1.0.1998
```

I have a personal account ending with @outlook.com account, and I'm able to access https://online.visualstudio.com/environments. Up until a couple days ago, I was running into the problem described in this issue. Now I'm able to authorize in the website, but nothing happens in my remote box.

*More details:*

I run `vso start`, and it shows a message directing to https://microsoft.com/devicelogin. While leaving the process running on the remote, I enter that address in my local Windows machine.
After entering the code and signing in, I get a successful message saying all is ready and that I can close the page. However, my remote has no reaction, `vso` is still hanging waiting for something.

I can see the process is connected to some external IP, so I guess it's still waiting for some information from the successful login:

```bash
$ sudo lsof -i -P -n
COMMAND   PID       USER   FD   TYPE     DEVICE SIZE/OFF NODE NAME
vso     28427       root  113u  IPv4 1234504388      0t0  TCP (my ip):50316->40.126.0.72:443 (ESTABLISHED)
```

It stays that way indefinitely.

I'm not sure if this is a different bug or a problem with my setup. Should I create a separate issue? Is there a way for me to debug this?

**Comment by Niek** [comment]
@gusbicalho same issue here, every 10 secs my syslog is spammed full of messages like this:
```bash
Apr 14 13:02:28 niek vso: CLI Information: 0 :
Apr 14 13:02:28 niek vso: Overriding serviceUri to https://prod.liveshare.vsengsaas.visualstudio.com/api/v1.2/
Apr 14 13:02:28 niek vsls-agent: Agent Information: 0 :
Apr 14 13:02:28 niek vsls-agent: Accepted incoming pipe connection.
Apr 14 13:02:28 niek vsls-agent: Agent Information: 0 :
Apr 14 13:02:28 niek vsls-agent: RPC session #8 established.
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent I] Accepted incoming pipe connection.
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vsls-agent: Agent-ObjectTracker Verbose: 0 :
Apr 14 13:02:28 niek vso: [Agent I] RPC session #8 established.
Apr 14 13:02:28 niek vsls-agent: Add item:RpcSession #8 count:2
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent-ObjectTracker V] Add item:RpcSession #8 count:2
Apr 14 13:02:28 niek vsls-agent: Agent Verbose: 0 :
Apr 14 13:02:28 niek vso: Client.Rpc Verbose: 0 :
Apr 14 13:02:28 niek vso: => ProcessMessagesInternalAsync session #:248
Apr 14 13:02:28 niek vso: Client.Rpc Verbose: 0 :
Apr 14 13:02:28 niek vso: < #248 Microsoft.Cascade.Rpc.RequestMessage
Apr 14 13:02:28 niek vsls-agent: => ProcessMessagesInternalAsync session #:8
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent V] => ProcessMessagesInternalAsync session #:8
Apr 14 13:02:28 niek vsls-agent: Agent Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: > #8 Microsoft.Cascade.Rpc.RequestMessage
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent V] > #8 {Id=1 Name=version.exchangeVersions Arguments=[null,{"extensionName":null,"extensionVersion":null,"applicationName":"vsonline-agent","applicationVersion":"1.0.1998","protocolVersion":null,"platformName":null,"platformVersion":null}]}
Apr 14 13:02:28 niek vsls-agent: Agent Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: < #8 Microsoft.Cascade.Rpc.ResponseMessage
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent V] < #8 {Result={VersionString=1.0.1998.63218 ProtocolVersionString=2.2 PlatformName=Linux 4.18.0 PlatformVersionString=4.18.0.25} Id=1}
Apr 14 13:02:28 niek vso: Client.Rpc Verbose: 0 :
Apr 14 13:02:28 niek vso: > #248 Microsoft.Cascade.Rpc.Json.JsonResponseMessage
Apr 14 13:02:28 niek vso: Client.Rpc Verbose: 0 :
Apr 14 13:02:28 niek vso: < #248 Microsoft.Cascade.Rpc.RequestMessage
Apr 14 13:02:28 niek vsls-agent: Agent Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: > #8 Microsoft.Cascade.Rpc.RequestMessage
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent V] > #8 {Id=2 Name=version.exchangeSettings Arguments=[{"agentAutoStart":true,"agentUri":"net.pipe://localhost/VSLS","serviceEndpoint":"https://prod.liveshare.vsengsaas.visualstudio.com/api/v1.2/","signalRServiceUri":"","isCloudEnvironment":true,"experimentalFeatures":{"propagateFolderEvents":false,"attemptReconnection":true},"increasedGuestLimit":true,"keychainFile":"/root/.vs-liveshare-keychain","syncRpcTelemetryTresholdMs":1000}]}
Apr 14 13:02:28 niek vsls-agent: Agent Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: < #8 Microsoft.Cascade.Rpc.ResponseMessage
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent V] < #8 {Result=<null> Id=2}
Apr 14 13:02:28 niek vso: Client.Rpc Verbose: 0 :
Apr 14 13:02:28 niek vso: > #248 Microsoft.Cascade.Rpc.Json.JsonResponseMessage
Apr 14 13:02:28 niek vso: Client.Rpc.Workspace Verbose: 0 :
Apr 14 13:02:28 niek vso: < #248 Microsoft.Cascade.Rpc.RequestMessage
Apr 14 13:02:28 niek vsls-agent: Agent.Rpc.Workspace Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: > #8 Microsoft.Cascade.Rpc.RequestMessage
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent.Rpc.Workspace V] > #8 {Id=3 Name=workspace.listWorkspaces Arguments=[{"includeRemoteWorkspaces":false}]}
Apr 14 13:02:28 niek vsls-agent: Agent.Rpc.Workspace Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: < #8 Microsoft.Cascade.Rpc.ResponseMessage
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent.Rpc.Workspace V] < #8 {Result={} Id=3}
Apr 14 13:02:28 niek vso: Client.Rpc.Workspace Verbose: 0 :
Apr 14 13:02:28 niek vso: > #248 Microsoft.Cascade.Rpc.Json.JsonResponseMessage
Apr 14 13:02:28 niek vso: Client.Rpc Information: 0 :
Apr 14 13:02:28 niek vso: RpcSession #248 is disposing.
Apr 14 13:02:28 niek vso: Client.Rpc Verbose: 0 :
Apr 14 13:02:28 niek vso: Emptying send queue of 0 messages.
Apr 14 13:02:28 niek vso: Client.Rpc Information: 0 :
Apr 14 13:02:28 niek vso: RpcSession #248 is disposing.
Apr 14 13:02:28 niek vso: Client.Rpc Verbose: 0 :
Apr 14 13:02:28 niek vso: <= ProcessMessagesInternalAsync session #:248
Apr 14 13:02:28 niek vsls-agent: Agent Information: 0 :
Apr 14 13:02:28 niek vsls-agent: #8 Disconnected: Reached end of stream.
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent I] #8 Disconnected: Reached end of stream.
Apr 14 13:02:28 niek vsls-agent: Agent Information: 0 :
Apr 14 13:02:28 niek vsls-agent: RPC session #8 disconnected: Reached end of stream.
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent I] RPC session #8 disconnected: Reached end of stream.
Apr 14 13:02:28 niek vsls-agent: Agent-ObjectTracker Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: Remove item:RpcSession #8 count:1 others:[KeepAliveAgent reason:AgentStarted delay:120000]
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent-ObjectTracker V] Remove item:RpcSession #8 count:1 others:[KeepAliveAgent reason:AgentStarted delay:120000]
Apr 14 13:02:28 niek vsls-agent: Agent Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: Emptying send queue of 0 messages.
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent V] Emptying send queue of 0 messages.
Apr 14 13:02:28 niek vsls-agent: Agent Information: 0 :
Apr 14 13:02:28 niek vsls-agent: RpcSession #8 is disposing.
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent I] RpcSession #8 is disposing.
Apr 14 13:02:28 niek vsls-agent: Agent Verbose: 0 :
Apr 14 13:02:28 niek vsls-agent: <= ProcessMessagesInternalAsync session #:8
Apr 14 13:02:28 niek vso: CLI Verbose: 2001 :
Apr 14 13:02:28 niek vso: [Agent V] <= ProcessMessagesInternalAsync session #:8
```

https://online.visualstudio.com/environments shows status "creating". Not sure if it's related in any way to the personal accounts (i.e. what this issue is about).

**Comment by davide-rota** [comment]
@Niek 
- `vso stop` (complete the process)
- remove all folders and files related to vsonline in your home folder.
- `sudo apt remove vso` -> reboot -> `sudo apt install vso
- `vso start`

Worked for me every time I messed up my installation!

**Comment by gusbicalho** [comment]
@davide-rota thanks for the suggestion, but I just tried that and it did not work =/

**Comment by taiyosogawa** [comment]
Hi all, quick update here:

The fix for the authentication error has been rolled out to some but all distribution points. Depending on your location and mode of install it may be another day or two until you see it show up.

@gusbicalho, thanks for the detailed info - it looks like you might be seeing #400, could you see if the workaround posted there unblocks you and open a new issue if it does not?

**Comment by taiyosogawa** [comment]
This should be resolved in all places now - if you're using the CLI directly, you may need to update it. If you're running via the extension, no update is required.

**Comment by alecloudenback** [comment]
I continue to get this error:
![image](https://user-images.githubusercontent.com/711879/89082889-f63b0c80-d354-11ea-8268-a0e55d41c43f.png)

**Comment by roflcoopter** [comment]
I also get this error suddenly

**Comment by davidwin93** [comment]
Im seeing this as well for a custom domain. I'm running the cli (codespaces v1.0.2579.56687 ) on Ubuntu 20.04.1

![error](https://user-images.githubusercontent.com/12397845/89852760-c78b1600-db4c-11ea-97f1-b07d33a9d792.png)

**Comment by alecloudenback** [comment]
@taiyosogawa worth re-opening this issue or should those of us with the issue open a new one?

**Comment by taiyosogawa** [comment]
Yes, let's re-open this issue and use it to track.

**Comment by timkeeler** [comment]
I'm having this issue with the updated `codespaces` command, let me know if I can provide additional troubleshooting. I'm unable to login with my personal account.

**Comment by timkeeler** [comment]
Does anyone happen to know of workarounds? I'm currently unable to use codespaces with self hosted instances.

**Comment by 2percentsilk** [comment]
As part of the [consolidation of VS Codespaces into GitHub](aka.ms/vscs-transition), self-hosted will need to be migrated to GitHub Codespaces. Since this migration isn't part of our roadmap and the self-hosted service will be retired on November 20, 2020, I'm going to close this as won't fix as we are not investing in this CLI anymore.

If you have any thoughts/feedback on self-hosted Codespaces support in GitHub, please let us know by commenting on this [issue](https://github.community/t/self-hosted-codespaces/131361/19).
